packages/video_player/video_player/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -404,17 +411,20 @@ - (void)dispose {
                                 context:playbackBufferFullContext];
   [_player replaceCurrentItemWithPlayerItem:nil];
   [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (void)dispose {
+  [self disposeSansEventChannel];
   [_eventChannel setStreamHandler:nil];
 }
 
 @end
 
 @interface FLTVideoPlayerPlugin ()
-@property(readonly, nonatomic) NSObject<FlutterTextureRegistry>* registry;
-@property(readonly, nonatomic) NSObject<FlutterBinaryMessenger>* messenger;
+@property(readonly, weak, nonatomic) NSObject<FlutterTextureRegistry>* registry;
+@property(readonly, weak, nonatomic) NSObject<FlutterBinaryMessenger>* messenger;
 @property(readonly, nonatomic) NSMutableDictionary* players;
-@property(readonly, nonatomic) NSObject<FlutterPluginRegistrar>* registrar;
-
+@property(readonly, weak, nonatomic) NSObject<FlutterPluginRegistrar>* registrar;
```

Comment:
who retains `registrar`?  in https://github.com/flutter/engine/blob/master/shell/platform/darwin/ios/framework/Source/FlutterEngine.mm#L666-L673
It seems like we autorelease the registrar.

Reply:
If you have the latest version of the engine, FlutterEngine holds on to the registrar.  If this is a strong property it would result in a cycle.  On old engines it wouldn't work.  We can either bump the required engine or adjust the FlutterEngine to not retain the registrars.

Reply:
Oh, never mind. I didn't see the line that engine holds the registrar. This is good! Thanks.

Reply:
Switched to a strong retain to be backwards compatible with older version of flutter.  The registrar has a weak link to the engine so it shouldn't block the deletion of the engine:
https://github.com/flutter/engine/blob/fdabcaddf45deaf93a5238c99c2acbeb25a3a027/shell/platform/darwin/ios/framework/Source/FlutterEngine.mm#L717